vtv-shows-app/src/app/page.tsx

```diff
@@ -0,0 +1,43 @@
+// pages/index.tsx
+'use client'
+import { Box, Heading } from "@chakra-ui/react";
+import ShowDetails from "../components/features/shows/ShowDetails";
+import ShowReviewSection from "../components/features/shows/ShowReviewSection";
+import { IShow, IReview } from "../typings/show";
+import styles from './page.module.css';
+
+const defaultShow: IShow = {
+    title: "Brooklyn Nine-Nine",
+    description: "Comedy series following the exploits of Det. Jake Peralta and his diverse, lovable colleagues as they police the NYPD's 99th Precinct.",
+    averageRating: 0,
+    imageUrl: "https://www.crimemuseum.org/wp-content/uploads/2014/05/Brooklyn-Nine-Nine.jpg"
+};
+
+let show: IShow = { ...defaultShow }; // Declare show here with default values
+
+const storedReviews = localStorage.getItem("reviews");
+if (storedReviews) {
+    const reviews: IReview[] = JSON.parse(storedReviews);
```

Comment:
What you are doing here is loading reviews from the local storage and calculating the average rating. This only happens on initial page load.

There is no mechanism that will rerender this page when a new review is added. That is why your average rating is never updated even after adding new reviews. You have to reload the full page to update the average rating.

What you should do is organise your components in a way that will keep the state above the components that need to update. So the state should be kept in a component that is a direct parent of both the reviews list and average rating.